taxonomy/api/v1/views.py

```diff
@@ -0,0 +1,68 @@
+"""
+Taxonomy API views.
+"""
+from rest_framework import permissions
+from rest_framework.mixins import ListModelMixin, RetrieveModelMixin
+from rest_framework.viewsets import GenericViewSet
+
+from django.db.models import Prefetch
+
+from taxonomy.api.v1.serializers import JobPostingsSerializer, JobsListSerializer, SkillListSerializer
+from taxonomy.models import CourseSkills, Job, JobPostings, Skill
+
+
+class TaxonomyAPIViewSetMixin:
+    """
+    Taxonomy APIs ViewSet Mixing.
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+    throttle_scope = 'taxonomy-api-throttle-scope'
+
+
+class SkillViewSet(TaxonomyAPIViewSetMixin, RetrieveModelMixin, ListModelMixin, GenericViewSet):
+    """
+    ViewSet to list and retrieve all Skills in the system.
+    """
+
+    serializer_class = SkillListSerializer
+
+    def get_queryset(self):
+        """
+        Get all the skills with prefetch_related objects
+        """
+        return Skill.objects.all().prefetch_related(
+            Prefetch(
+                'courseskills_set',
+                CourseSkills.objects.filter(is_blacklisted=False)
+            )
+        )
+
+
+class JobsViewSet(TaxonomyAPIViewSetMixin, RetrieveModelMixin, ListModelMixin, GenericViewSet):
+    """
+    ViewSet to list and retrieve all Jobs in the system.
+    """
+    serializer_class = JobsListSerializer
+
+    def get_queryset(self):
+        """
+        Get all the jobs with prefetch_related objects
+        """
+        return Job.objects.all().prefetch_related(
```

Comment:
What is the query count with and without prefetch?

Reply:
for my local data, 6 querys with prefetch and 40 queries without it.

Reply:
how are you testing the query counts? `Django Debug Toolbar`?

Reply:
Yes Django Debug Toolbar